api/controllers/ViewController.js

```diff
@@ -593,10 +648,42 @@ module.exports = {
   async user(request, reply) {
     // If the user is not authenticated, redirect to the login page
     const cookie = request.yar.get('session');
-    if (!cookie || (cookie && !cookie.userId) || (cookie && !cookie.totp)) {
+    if (!cookie || !cookie.userId || !cookie.totp) {
+      logger.info(
+        '[ViewController->user] User had no session. Redirecting to login.',
+        {
+          request,
+        },
+      );
```

Comment:
DRY? Maybe calling `noSessionRedirectsToLogin` ? :)

Reply:
I agree with you and tried to avoid it. Problem is, I had difficulty using `reply` to act on the conditional if it's inside another function. So basically it only removes part of the boilerplate but you're still left with:

```js
if (!hasValidSession(request, 'METHOD_NAME')) {
  return reply.redirect('/');
}

// Now collect cookie data AGAIN so this function can use it.
const cookie = request.yar.get('session');

... stuff ...
```

And this results in some "HID Classic" lie-prone logging, where the actual log entry appears BEFORE you redirect, and it would be possible to not even do the redirect, like so:

```js
if (!hasValidSession(request, 'METHOD_NAME')) {
  // I'm not redirecting, but `hasValidSession` already created a log entry saying I was gonna do that
  return reply.view('error');
}
```

Additionally, while tinkering I couldn't find a nice way to make the logs record which function was punting the user back to login. I tried looking at automatic stuff like `args.caller` or even a separate param on the function but I could never find the "magic" I wanted. So the examples above have `'METHOD_NAME` as a required param in the hypothetical `hasValidSession()` function, so that they can log on behalf of said method. Again, kinda backwards.

Anyway, that's why I copy/pasted the code into each method. I will gladly take suggestions if all the above issues can be considered!

Reply:
Yeah, it is probably not worth it all things considered.